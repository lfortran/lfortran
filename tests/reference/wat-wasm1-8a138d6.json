{
    "basename": "wat-wasm1-8a138d6",
    "cmd": "lfortran --no-color --show-wat {infile}",
    "infile": "tests/wasm1.f90",
    "infile_hash": "6027c04f5ca0464227bf7443522c5d118cadf6e4404c9224bbc63d7b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "wat-wasm1-8a138d6.stdout",
    "stdout_hash": "f4d22c552709850528ab576438381b829d054835779927ae87bb778b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}