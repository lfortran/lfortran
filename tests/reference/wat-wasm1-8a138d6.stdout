(module
    (type (;0;) (func (param i32) (result)))
    (type (;1;) (func (param i64) (result)))
    (type (;2;) (func (param f32) (result)))
    (type (;3;) (func (param f64) (result)))
    (type (;4;) (func (param i32 i32) (result)))
    (type (;5;) (func (param) (result)))
    (type (;6;) (func (param i32) (result i32)))
    (type (;7;) (func (param i32 i32) (result i32)))
    (type (;8;) (func (param i64 i64) (result i64)))
    (type (;9;) (func (param i32) (result i32)))
    (type (;10;) (func (param i32 i32) (result i32)))
    (type (;11;) (func (param) (result)))
    (import "js" "print_i32" (func (;0;) (type 0)))
    (import "js" "print_i64" (func (;1;) (type 1)))
    (import "js" "print_f32" (func (;2;) (type 2)))
    (import "js" "print_f64" (func (;3;) (type 3)))
    (import "js" "print_str" (func (;4;) (type 4)))
    (import "js" "flush_buf" (func (;5;) (type 5)))
    (import "js" "memory" (memory (;0;) 10 100))
    (func $6 (type 6) (param i32) (result i32)
        (local i32)
        local.get 0
        local.get 0
        i32.mul
        local.set 1
        local.get 1
        return
    )
    (func $7 (type 7) (param i32 i32) (result i32)
        (local i32)
        local.get 0
        local.get 1
        i32.add
        local.set 2
        local.get 2
        return
    )
    (func $8 (type 8) (param i64 i64) (result i64)
        (local i64)
        local.get 0
        local.get 1
        i64.add
        local.set 2
        local.get 2
        return
    )
    (func $9 (type 9) (param i32) (result i32)
        (local i32 i32)
        i32.const 3
        local.set 2
        local.get 2
        local.get 0
        call 6
        i32.mul
        local.set 1
        local.get 1
        return
    )
    (func $10 (type 10) (param i32 i32) (result i32)
        (local i32)
        local.get 0
        local.get 1
        i32.add
        local.set 2
        local.get 2
        return
    )
    (func $11 (type 11) (param) (result)
        (local)
        i32.const 5
        call 6
        call 0
        call 5
        i32.const 5
        i32.const 4
        call 7
        call 0
        call 5
        i64.const 4
        i64.const 5
        call 8
        call 1
        call 5
        i32.const 5
        call 9
        call 0
        call 5
        i32.const 5
        i32.const 4
        call 10
        call 0
        call 5
        return
    )
    (export "a_sqr" (func $6))
    (export "add" (func $7))
    (export "add64" (func $8))
    (export "computecirclearea" (func $9))
    (export "my_add" (func $10))
    (export "_lcompilers_main" (func $11))
)
